lib/Doctrine/Common/Persistence/Mapping/AbstractClassMetadataFactory.php

```diff
@@ -398,16 +395,28 @@ public function getReflectionService()
     }
 
     /**
-     * Gets the real class name of a class name that could be a proxy.
+     * Gets the real class name of a class name that could be a proxy or alias.
      */
-    private function getRealClass(string $class) : string
+    protected function getRealClassName(string $className) : string
     {
-        $pos = strrpos($class, '\\' . Proxy::MARKER . '\\');
+        if (isset($this->aliasesMap[$className])) {
```

Comment:
Considering that duplicate entries are allowed, is this property still needed?

Reply:
This was done because of https://github.com/doctrine/persistence/pull/21#discussion_r235214384 Can you please decipher comment chain and explain if something still needs to be done?